cleanlab/typing.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Union
+import numpy as np
+import pandas as pd
+
+LabelLike = Union[list, np.ndarray, pd.Series, pd.DataFrame]
```

Comment:
Can we type these as type aliases? (https://github.com/cleanlab/cleanlab/pull/317#discussion_r927905542)

Reply:
Only for Python 3.10.

Reply:
https://docs.python.org/3/library/typing.html#typing.TypeAlias

Reply:
```python
>>> import sys
>>> sys.version
'3.8.5 (default, Sep  4 2020, 07:30:14) \n[GCC 7.3.0]'
>>> from typing import TypeAlias
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'TypeAlias' from 'typing'
```

Reply:
@anishathalye Did you have something like this in mind?

```python
# Import TypeAlias on Python 3.10+
try:
    from typing import TypeAlias
except ImportError:
    TypeAlias = type(None)


LabelLike: TypeAlias = Union[list, np.ndarray, pd.Series, pd.DataFrame]
DatasetLike: TypeAlias = Any
```

Reply:
Ah I didn't realize that was only added in 3.10. We want to support 3.6+ without too much compatibility code, so let's stick with the current code in the PR.